lib/ex_unit/lib/ex_unit/failures_manifest.ex

```diff
@@ -0,0 +1,100 @@
+defmodule ExUnit.FailuresManifest do
+  @moduledoc false
+
+  @type test_id :: {module, name :: atom}
+  @opaque t :: %{test_id => test_file :: Path.t()}
+
+  @manifest_vsn 1
+
+  @spec new() :: t
+  def new, do: %{}
+
+  @spec files_with_failures(t) :: MapSet.t(Path.t())
+  def files_with_failures(%{} = manifest) do
+    manifest
+    |> Map.values()
+    |> MapSet.new()
+  end
+
+  @spec failed_test_ids(t) :: MapSet.t(test_id)
+  def failed_test_ids(%{} = manifest) do
+    manifest
+    |> Map.keys()
+    |> MapSet.new()
+  end
+
+  @spec put_test(t, ExUnit.Test.t()) :: t
```

Comment:
Let's add a TODO here so we don't forget to check why we need this clause after merging. :)

Reply:
Done.